enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,901 @@
+        },
+        {
+        "name": "olm.channel",
+        "value": "singlenamespace-alpha"
+        },
+        {
+            "name": "olm.channel",
+            "value": "clusterwide-alpha"
+        }
+    ],
+    "relatedImages" : [
+        {
+            "name": "etcdv0.9.0",
+            "image": "quay.io/coreos/etcd-operator@sha256:db563baa8194fcfe39d1df744ed70024b0f1f9e9b55b5923c2f3a413c44dc6b8"
+        }
+    ]
+},
+.
+.
+.
+.
+```
+
+Similarly, removing a `.json` from `community-operators` and pushing that update up to the container image removes the operator from the index. 
+
+### Story 3
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can validate the structure of the content of an index using the json representation.
+ 
+```bash
+$ cd community-operators 
+$ sed '/defaultChannel/d' etcd.json > etcd.json //delete the line "defaultChannel":"singlenamespace-alpha"
+$ git status 
+	modified:    etcd.json
+
+$ cd ..
+$ opm index validate community-operators
+marshal error: etcd.defaultchannel: cannot convert incomplete value "string" to JSON
+$ opm index update community-operator --tag=docker.io/mynamespace/community-operators:latest
+$ opm index validate docker.io/mynamespace/community-operators:latest 
+marshal error: etcd.defaultchannel: cannot convert incomplete value "string" to JSON
+$ cd community-operators 
+$ git checkout etcd.json
+$ cd ..
+$ opm index validate community-operators 
+No errors found!
+```
+
+### Story 4 
+
+Given a set of package representations, I can author a new index using just those representations.
+
+```bash
+$ tree community-operators 
+community-operators
+├── amqstreams.json
+└── etcd.json
+└── servicemesh.json
+$ rm community-operators/servicemesh.json
+$ opm index create --from=community-operators --tag=docker.io/someothernamespace/new-community-operators:latest
+```
+## Implementation Details
+### Representing a package using json/yaml 
+
+The representation of the content of an index using json/yaml provides a unique opportunity to rethink the structure of an index conceptually. 
+
+![Alt text](assets/new-community-operators.png?raw=true "community-operators")
+
+Since each bundle in the index belongs to a channel in a package, the index itself can be represented as a collection of packages. Each package is then represented as a json/yaml, a config file that will live inside the index container image alongside the individual bundle metadata. 
+
+#### Contents of the config file
+
+The config file for each package will have a stream of json objects, representing the package and bundle information for that package. Currently, this information is stored in a sql database inside the index. To capture all the information previously stored in this database but in a normalized way, the config file will have two kinds of json blob: 
+
+1. The json blob capturing package information. 
+```json
+{
+    "schema": "olm.package",
```

Comment:
Should we include schema version identifier here as well?

Reply:
an older version did, but I think we can say that unadorned schema is v1 and if we need a v2 it will be "olm.package.v2"

(the fact we have an extension mechanism via properties means we may never need to bump a version)